backend/controller/cronjobs/state.go

```diff
@@ -0,0 +1,82 @@
+package cronjobs
+
+import (
+	"time"
+
+	"github.com/TBD54566975/ftl/backend/controller/dal"
+	"github.com/TBD54566975/ftl/internal/model"
+	"github.com/TBD54566975/ftl/internal/slices"
+	"github.com/alecthomas/types/optional"
+)
+
+type State struct {
+	jobs []dal.CronJob
```

Comment:
I was thinking a bit more about the other PR where I asked you to make the `(Module, Verb)` field into a `schema.Ref`, and I think what my brain was trying to tell me is that I think we should be separating the layers more here - rather than using types from the DAL such as `dal.CronJob`, we should be adding abstract model types to [internal/model](https://github.com/TBD54566975/ftl/blob/main/internal/model/model.go) and using those. I'm thinking of a `CronJob` type and a `VerbRef` type.

I'm not 100% sure that makes sense though, WDYT?

Reply:
On second thought, I think what I'm suggesting is that we move more of the data structures from the `dal` package into the `internal/model` package, which is outside the scope of this PR.

Reply:
Yeh I was weighing up dal vs model package when making the type as well.
Originally I liked having the type in dal because the primary id could remain private. But thats been solved with the key field now.
I agree on moving more types out of dal and into model. I'll do it for the new types (CronJob + VerbRef) but leave existing ones alone